### ISSUE TITLE
Valsets members with Zero/Empty Eth Address

### ISSUE THREAD

**Issue by mankenavenkatesh** [althea-net/cosmos-gravity-bridge#271]
**Context**

While creating Valset request, we iterate through bonded validators, do a lookup to fetch Ethereum Address and 

- if Ethereum address is found, the bridge validator is created with Ethereum Address.
- If Ethereum Address is not found, the bridge validator is created with default address. (Zero Address)

Reference - 
https://github.com/althea-net/cosmos-gravity-bridge/blob/b89887d519e0159ae7c426c5c5cb3fb739242835/module/x/gravity/keeper/keeper.go#L439

**Problem Statement:**
Based on above logic, new valset request can be created with empty Eth addresses and with 2/+1 confirmations from previous valset members, the new valset gets updated on Gravity contract eventually. 

If the power of empty addresses is > 2/3+1, the next set of valsets can never be submitted to Gravity contract due to insufficient voting power.

**Example:**
 Consider the following scenario
Let's say Gravity contract is initialised with following valset.
```
nonce:0 members:<power:4294967295 ethereum_address:"0x50fC936Df8Ed96e6c04891df421c4655eDFe5A64">
```

Once cosmos chain is started, Let's say following valset request is created automatically. (validator did not register his eth key yet.) and confirmed by previous valset members, and eventually submitted to Gravity contract.
```
nonce:1 members:<power:4294967295 ethereum_address:"0x0000000000000000000000000000000000000000"> 

```

Now validator registers his eth key. On cosmos, new valset request with nonce=2 is created.
```
nonce:2 members:<power:4294967295 ethereum_address:"0x50fC936Df8Ed96e6c04891df421c4655eDFe5A64"> 
```

Valset with nonce=2 is blocked. It can never be submitted to Gravity contract due to insufficient voting power.

**Solution**
When creating valset request, ignore validators who didn't register the eth key. 
So that valset request members don't have Zero Eth addresses.

**Comment by jkilpatr** [comment]
You're correct that this is an issue. It's an edge case we haven't really handled.

Just removing them from the list won't help, we would have to increase their relative voting power of the remaining validator s as well. For example if we just exlude 66% of the validator set that has not registered a key then we won't be able to pass anything else.

**Comment by mankenavenkatesh** [comment]
Agreed.
we should not consider zero addresses as members of valset request. 
we should filter non zero addresses and take only their powers into consideration for calculating total power and normalised powers.

**Comment by jkilpatr** [comment]
I have a fix running through tests locally. it's pretty small.

**Comment by mankenavenkatesh** [comment]
LGTM
